src/autocomplete/QueryMatcher.ts

```diff
@@ -107,35 +106,40 @@ export default class QueryMatcher<T extends Object> {
         if (query.length === 0) {
             return [];
         }
-        const results = [];
+        const matches = [];
         // Iterate through the map & check each key.
         // ES6 Map iteration order is defined to be insertion order, so results
         // here will come out in the order they were put in.
-        for (const key of this._items.keys()) {
-            let {value: resultKey} = key;
+        for (const [key, candidates] of this._items.entries()) {
+            let resultKey = key;
             if (this._options.shouldMatchWordsOnly) {
                 resultKey = resultKey.replace(/[^\w]/g, '');
             }
             const index = resultKey.indexOf(query);
             if (index !== -1 && (!this._options.shouldMatchPrefix || index === 0)) {
-                results.push({key, index});
+                matches.push(
+                    ...candidates.map((candidate) => ({key, index, ...candidate}))
```

Comment:
not that it matters much but do we need to put `key` inside matches? it seems to only use `index`, `keyWeight` + `object` both which come from `...candidate`

Reply:
Good catch! I've removed it